osquery/tables/applications/posix/docker.cpp

```diff
@@ -563,11 +563,22 @@ QueryData genContainerProcesses(QueryContext& context) {
     }
 
     pt::ptree container;
-    Status s =
-        dockerApi("/containers/" + id + "/top?ps_args=axwwo%20" +
-                      "pid,state,uid,gid,euid,egid,suid,sgid,rss,vsz,etimes,"
-                      "ppid,pgrp,nlwp,nice,user,time,pcpu,pmem,comm,cmd",
-                  container);
+
+#if defined(__APPLE__)
+    // osx: 19 fields
+    const std::string ps_args =
+        "pid,state,uid,gid,svuid,svgid,rss,vsz,etime,ppid,pgid,wq,nice,user,"
```

Comment:
Why bother querying all of these if they are not used below? It looks like you only are using PID, UID, TIME and CMDLINE.

Reply:
Those fields are in preparation for the eventual support of Docker to return values for all the fields on OS X. Currently no matter what one provides to the Docker API for ps_args, you get PID, UID, TIME and CMDLINE.

Reply:
👍 Helpfully docker guys will implement this properly instead of relying on container ps command!!!